op-deployer/pkg/deployer/artifacts/embedded.go

```diff
@@ -96,3 +95,18 @@ func ExtractFromFile(destDir string, tarFilePath string) (foundry.StatDirFs, err
 
 	return os.DirFS(forgeArtifactsDir).(foundry.StatDirFs), nil
 }
+
+func ExtractArtifactsToTemp() (bundleRoot string, err error) {
+	tmpDir, err := os.MkdirTemp("", "forge-extraction-")
+	if err != nil {
+		return "", fmt.Errorf("failed to create temp dir: %w", err)
+	}
+
+	_, bundleRoot, err = ExtractEmbedded(tmpDir)
+	if err != nil {
+		os.RemoveAll(tmpDir)
+		return "", err
+	}
+
+	return bundleRoot, nil
+}
```

Comment:
Could we remove this helper function and just pass a tempDir directly to the existing ExtractEmbedded? Then the caller could handle cleanup.
```
ExtractEmbedded(tmpDir)
```
As currently written you create a tempDir but if there are no errors, it never gets cleaned up